src/main/java/ch/njol/skript/variables/TypeHints.java

```diff
@@ -51,7 +54,20 @@ public static void add(String variable, Class<?> hint) {
 		Map<String, Class<?>> hints = typeHints.getFirst();
 		hints.put(variable, hint);
 	}
-	
+
+	/**
+	 * Return any known type hints of a local variable.
+	 * 
+	 * @param variable The local variable expression to check against.
+	 * @return The return type that the local variable has been set to otherwise null if unset.
+	 */
+	@Nullable
+	public static Class<?> get(Variable<?> variable) {
+		if (!variable.isLocal())
+			throw new SkriptAPIException("Must only get TypeHints of local variables.");
+		return get(variable.getName().toString());
```

Comment:
How does this work for variable names that have expressions?

Reply:
It doesn't. It must be simple.

Reply:
Sounds good

Reply:
It could though. I did fix the memory system for variable strings in my default variables pull request.

This is how it was prior. It's a bad idea for these type hints though because then it'll store every local variable type in ram, especially bad if there is a while loop and it's generating numbers as the index. If it's simple, someone would have to write a 1 million long line script for it to start become an issue.